backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -188,74 +252,32 @@ def getResourceInst(self, path: str, environ: dict):
+        tracim_context = environ["tracim_context"]
         tracim_context.set_path(path)
-        user = tracim_context.current_user
-        session = tracim_context.dbsession
-        if not self.exists(path, environ):
-            return None
-        root_path = tracim_context.root_path
-
-        # If the requested path is the root, then we return a RootResource resource
-        if path == root_path:
+        current_path_level = tracim_context.processed_path.current_path_level
+        # root
+        if current_path_level == ResourceLevelType.ROOT:
             return resources.RootResource(path=path, environ=environ, tracim_context=tracim_context)
-
-        try:
+        if current_path_level == ResourceLevelType.WORKSPACE:
             workspace = tracim_context.current_workspace
-        except WorkspaceNotFound:
-            workspace = None
-
-        # If the request path is in the form root/name, then we return a WorkspaceResource resource
-        parent_path = dirname(path)
-        if parent_path == root_path:
-            if not workspace:
-                return None
-            return resources.WorkspaceResource(
+            return get_workspace_resource(
                 path=path,
                 environ=environ,
                 workspace=workspace,
                 tracim_context=tracim_context,
-                label=workspace.label,
+                label=workspace.filemanager_filename,
             )
-
-        # And now we'll work on the path to establish which type or resource is requested
-
-        ContentApi(
-            current_user=user,
-            session=session,
-            config=self.app_config,
-            show_archived=False,
-            show_deleted=False,
-            namespaces_filter=[ContentNamespaces.CONTENT],
-        )
-
-        try:
+        if current_path_level == ResourceLevelType.CONTENT:
             content = tracim_context.current_content
-        except ContentNotFound:
-            content = None
-
-        # And if we're still going, the client is asking for a standard Folder/File/Page/Thread so we check the type7
-        # and return the corresponding resource
-
-        if content is None:
-            return None
-        if content.type == content_type_list.Folder.slug:
-            return resources.FolderResource(
+            return get_content_resource(
                 path=path,
                 environ=environ,
                 workspace=content.workspace,
                 content=content,
                 tracim_context=tracim_context,
             )
-        elif content.type == content_type_list.File.slug:
-            return resources.FileResource(
-                path=path, environ=environ, content=content, tracim_context=tracim_context,
-            )
-        else:
-            return resources.OtherFileResource(
-                path=path, environ=environ, content=content, tracim_context=tracim_context,
-            )
+        return None
```

Comment:
Can you add the typing information on this method?

Reply:
done.